packages/vscode-extension/src/exp/index.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import { IExperimentationService, IExperimentationTelemetry, TargetPopulation, getExperimentationServiceAsync } from "vscode-tas-client";
+import * as pkg from "./../../package.json";
+import { ExtTelemetry } from "./../telemetry/extTelemetry"
+
+export namespace exp {
+    let expService: IExperimentationService;
+
+    export function getExpService(): IExperimentationService {
+        return expService;
+    }
+
+    export async function initialize(context: vscode.ExtensionContext) {
+        expService = await getExperimentationServiceAsync(
+            pkg.name,
+            pkg.version,
+            TargetPopulation.Public,
+            new ExperimentationTelemetry(),
+            context.globalState
+        );
+    }
+
+    class ExperimentationTelemetry implements IExperimentationTelemetry {
+        public setSharedProperty(name: string, value: string): void {
+            ExtTelemetry.addSharedProperty(name, value);
+        }
+    
+        public postEvent(eventName: string, props: Map<string, string>): void {
+            let properties: {[key: string]: string} = {};
+            props.forEach((value, key) => {
+                properties[key] = value;
+            });
+            ExtTelemetry.sendTelemetryEvent(eventName, properties);
```

Comment:
One question, the props here are not the shared properties that you attached  to vscode telemetry reporter?

Reply:
here is the example of the props, other shared properties will be added by the reporter.
![image](https://user-images.githubusercontent.com/71362691/123030089-c7b69c00-d414-11eb-8e84-ec201c28d7f5.png)